Lib/glyphsLib/classes.py

```diff
@@ -1477,37 +1477,62 @@ def parent(self):
 
 class GSFontMaster(GSBase):
     def _serialize_to_plist(self, writer):
-        writer.writeObjectKeyValue(self, "alignmentZones", "if_true")
-        writer.writeObjectKeyValue(self, "ascender")
-        writer.writeObjectKeyValue(self, "capHeight")
-        if self.customName:
-            writer.writeKeyValue("custom", self.customName)
-        writer.writeObjectKeyValue(self, "customValue", "if_true")
-        writer.writeObjectKeyValue(self, "customValue1", "if_true")
-        writer.writeObjectKeyValue(self, "customValue2", "if_true")
-        writer.writeObjectKeyValue(self, "customValue3", "if_true")
+        if writer.format_version == 2:
+            writer.writeObjectKeyValue(self, "alignmentZones", "if_true")
+            writer.writeObjectKeyValue(self, "ascender")
+        if writer.format_version == 3 and self.axes:
+            writer.writeKeyValue("axesValues", self.axes)
+        if writer.format_version == 2:
+            writer.writeObjectKeyValue(self, "capHeight")
+            if self.customName:
+                writer.writeKeyValue("custom", self.customName)
+            writer.writeObjectKeyValue(self, "customValue", "if_true")
+            writer.writeObjectKeyValue(self, "customValue1", "if_true")
+            writer.writeObjectKeyValue(self, "customValue2", "if_true")
+            writer.writeObjectKeyValue(self, "customValue3", "if_true")
+
         writer.writeObjectKeyValue(self, "customParameters", "if_true")
-        writer.writeObjectKeyValue(self, "descender")
+
+        if writer.format_version == 2:
+            writer.writeObjectKeyValue(self, "descender")
 
         if self.guides:
-            writer.writeKeyValue("guideLines", self.guides)
+            if writer.format_version == 3:
+                writer.writeKeyValue("guides", self.guides)
+            else:
+                writer.writeKeyValue("guideLines", self.guides)
 
-        writer.writeObjectKeyValue(self, "horizontalStems", "if_true")
+        if writer.format_version == 2:
+            writer.writeObjectKeyValue(self, "horizontalStems", "if_true")
 
         writer.writeObjectKeyValue(self, "iconName", "if_true")
         writer.writeObjectKeyValue(self, "id")
         writer.writeObjectKeyValue(self, "italicAngle", "if_true")
+        if writer.format_version == 3:
+            writer.writeKeyValue("metricValues", self.metrics)
 
-        if self._name and self._name != self.name:
+        if (
+            self._name and self._name != self.name
+        ) or writer.format_version == 3:  # Check this
```

Comment:
Hm?

Reply:
I've checked it, it's true! (Glyphs 2 omits the master name if it's "Regular", Glyphs 3 always writes it.)
We can remove that comment.